ERCS/erc-7943.md

```diff
@@ -0,0 +1,501 @@
+    /// This overwrites the current value, similar to an `approve` function.
+    /// @dev Requires specific authorization. Frozen tokens cannot be transferred by the user.
+    /// @param user The address of the user whose tokens are to be frozen/unfrozen.
+    /// @param tokenId The ID of the token to freeze/unfreeze.
+    /// @param amount The amount of tokens to freeze/unfreeze. 
+    function setFrozen(address user, uint256 tokenId, uint256 amount) external;
+
+    /// @notice Checks the frozen status/amount of a specific `tokenId`.
+    /// @param user The address of the user.
+    /// @param tokenId The ID of the token.
+    /// @return amount The amount of `tokenId` tokens currently frozen for `user`.
+    function getFrozen(address user, uint256 tokenId) external view returns (uint256 amount);
+ 
+    /// @notice Checks if a transfer is currently possible according to token rules. It enforces validations on the frozen tokens.
+    /// @dev This may involve checks like allowlists, blocklists, transfer limits and other policy-defined restrictions.
+    /// @param from The address sending tokens.
+    /// @param to The address receiving tokens. 
+    /// @param tokenId The ID of the token being transferred.
+    /// @param amount The amount being transferred.
+    /// @return allowed True if the transfer is allowed, false otherwise.
+    function isTransferAllowed(address from, address to, uint256 tokenId, uint256 amount) external view returns (bool allowed);
+
+    /// @notice Checks if a specific user is allowed to interact according to token rules.
+    /// @dev This is often used for allowlist/KYC/KYB/AML checks.
+    /// @param user The address to check.
+    /// @return allowed True if the user is allowed, false otherwise.
+    function isUserAllowed(address user) external view returns (bool allowed);
+}
+```
+
+### `isUserAllowed`, `isTransferAllowed` and `getFrozen`
+
+These provide views into the implementing contract's compliance, transfer policy logic and freezing status. These functions:
+
+- MUST NOT revert. 
+- MUST NOT change the storage of the contract. 
+- MAY depend on context (e.g., current timestamp, block number).
+- The `isTransferAllowed` MUST validate that the `amount` being transferred doesn't exceed the unfrozen amount (which is the difference between the current balance and the frozen balance).
+
+### `forceTransfer`
+
+This function provides a standard mechanism for forcing a transfer from a `from` to a `to` address. The function:
+
+- MUST directly manipulate balances or ownership to transfer the asset from `from` to `to` either by transferring or burning from `from` and minting to `to`.
+- MUST be restricted in access.
+- MUST perform necessary validation checks (e.g., sufficient balance/ownership of a specific token).
+- MUST emit both the standard `Transfer` event (from the base standard) and the `ForcedTransfer` event.
```

Comment:
Nitpick here, but I'd go for a different display of the underlying transfer event across the document as ERC20 and ERC721 share the `Transfer` name, but ERC1155 does not, with `TransferSingle` and `TransferBatch`.

I'd reference the event without quoting and maybe I'd add a one-liner explanation in the additional specifications section.

```suggestion
- MUST emit both the standard transfer event (from the base standard) and the `ForcedTransfer` event.
```